packages/mdc-toolbar/mdc-toolbar.scss

```diff
@@ -68,6 +68,7 @@ $mdc-toolbar-mobile-breakpoint: 599px;
     &--align-end {
```

Comment:
Instead of modifying `--align-end` directly, we would like to have a `mdc-toolbar--section-shrink-to-fit` modifier that set `flex:0`. 

In your specific use case, you can do something like:
```
<section class="mdc-toolbar__section mdc-toolbar__section--align-start">
  <span class="mdc-toolbar__title">This is a very long title</span>
</section>
<section class="mdc-toolbar__section mdc-toolbar__section--align-end mdc-toolbar--section-shrink-to-fit">
  ...
</section>
```